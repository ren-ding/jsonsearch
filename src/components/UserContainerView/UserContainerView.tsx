import React, { Component } from 'react';
import UserContainer from '../../model/UserContainer';
import ItemView from '../ItemView';

export interface UserContainerViewProps {
    userContainer: UserContainer
}

export default class UserContainerView extends Component<UserContainerViewProps> {
    constructor(props:UserContainerViewProps) {
      super(props);
    }
  
    render() {
      const displayOrganization = (this.props.userContainer.organization != null);

      return (
        <div>
          <h2>User:</h2>
            <ItemView
              itemObject ={this.props.userContainer.user}
            />
          <h2>Organization:</h2>
            <ItemView
              itemObject ={this.props.userContainer.organization || {}}
            />
          <h2>SubmitTickets:</h2>
            {this.props.userContainer.submitTickets.map(ticket => (
              <ItemView
                itemObject ={ticket}
              />
            ))}
          <h2>AssignedTickets:</h2>
            {this.props.userContainer.assignedTickets.map(ticket => (
              <ItemView
                itemObject ={ticket}
              />
            ))}
        </div>
      );
      
    }
}