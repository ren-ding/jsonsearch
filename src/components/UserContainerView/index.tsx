import UserContainerView from './UserContainerView';
export default UserContainerView;