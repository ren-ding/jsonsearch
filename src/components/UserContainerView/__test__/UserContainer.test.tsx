import React from 'react';
import UserContainerView, {UserContainerViewProps} from '../UserContainerView';
import renderer from 'react-test-renderer';

describe('UserContainerView', () => {
    let props:UserContainerViewProps;
    beforeEach(()=> {
        props = {
            userContainerList: [{
                user: {
                    "_id": 1,
                    "url": "http://initech.zendesk.com/api/v2/users/1.json",
                    "external_id": "74341f74-9c79-49d5-9611-87ef9b6eb75f",
                    "name": "Francisca Rasmussen",
                    "alias": "Miss Coffey",
                    "created_at": "2016-04-15T05:19:46 -10:00",
                    "active": true,
                    "verified": true,
                    "shared": false,
                    "locale": "en-AU",
                    "timezone": "Sri Lanka",
                    "last_login_at": "2013-08-04T01:03:27 -10:00",
                    "email": "coffeyrasmussen@flotonic.com",
                    "phone": "8335-422-718",
                    "signature": "Don't Worry Be Happy!",
                    "organization_id": 119,
                    "tags": [
                      "Springville",
                      "Sutton",
                      "Hartsville/Hartley",
                      "Diaperville"
                    ],
                    "suspended": true,
                    "role": "admin"
                  },
                organization: {
                    "_id": 101,
                    "url": "http://initech.zendesk.com/api/v2/organizations/101.json",
                    "external_id": "9270ed79-35eb-4a38-a46f-35725197ea8d",
                    "name": "Enthaze",
                    "domain_names": [
                      "kage.com",
                      "ecratic.com",
                      "endipin.com",
                      "zentix.com"
                    ],
                    "created_at": "2016-05-21T11:10:28 -10:00",
                    "details": "MegaCorp",
                    "shared_tickets": false,
                    "tags": [
                      "Fulton",
                      "West",
                      "Rodriguez",
                      "Farley"
                    ]
                  },
                submitTickets: [],
                assignedTickets: []
            }]
        }
    });

    it('should render to match snapshot', () => {
        const tree = renderer.create(<UserContainerView {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});