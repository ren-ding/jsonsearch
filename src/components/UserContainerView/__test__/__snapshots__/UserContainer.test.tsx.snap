// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContainerView should render to match snapshot 1`] = `
<div>
  <h2>
    User:
  </h2>
  <div>
    <table
      className="ui compact table"
    >
      <thead>
        <tr>
          <th>
            Field
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            _id
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td>
            url
          </td>
          <td>
            http://initech.zendesk.com/api/v2/users/1.json
          </td>
        </tr>
        <tr>
          <td>
            external_id
          </td>
          <td>
            74341f74-9c79-49d5-9611-87ef9b6eb75f
          </td>
        </tr>
        <tr>
          <td>
            name
          </td>
          <td>
            Francisca Rasmussen
          </td>
        </tr>
        <tr>
          <td>
            alias
          </td>
          <td>
            Miss Coffey
          </td>
        </tr>
        <tr>
          <td>
            created_at
          </td>
          <td>
            2016-04-15T05:19:46 -10:00
          </td>
        </tr>
        <tr>
          <td>
            active
          </td>
          <td>
            true
          </td>
        </tr>
        <tr>
          <td>
            verified
          </td>
          <td>
            true
          </td>
        </tr>
        <tr>
          <td>
            shared
          </td>
          <td>
            false
          </td>
        </tr>
        <tr>
          <td>
            locale
          </td>
          <td>
            en-AU
          </td>
        </tr>
        <tr>
          <td>
            timezone
          </td>
          <td>
            Sri Lanka
          </td>
        </tr>
        <tr>
          <td>
            last_login_at
          </td>
          <td>
            2013-08-04T01:03:27 -10:00
          </td>
        </tr>
        <tr>
          <td>
            email
          </td>
          <td>
            coffeyrasmussen@flotonic.com
          </td>
        </tr>
        <tr>
          <td>
            phone
          </td>
          <td>
            8335-422-718
          </td>
        </tr>
        <tr>
          <td>
            signature
          </td>
          <td>
            Don't Worry Be Happy!
          </td>
        </tr>
        <tr>
          <td>
            organization_id
          </td>
          <td>
            119
          </td>
        </tr>
        <tr>
          <td>
            tags
          </td>
          <td>
            Springville,Sutton,Hartsville/Hartley,Diaperville
          </td>
        </tr>
        <tr>
          <td>
            suspended
          </td>
          <td>
            true
          </td>
        </tr>
        <tr>
          <td>
            role
          </td>
          <td>
            admin
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h2>
    Organization:
  </h2>
  <div>
    <table
      className="ui compact table"
    >
      <thead>
        <tr>
          <th>
            Field
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            _id
          </td>
          <td>
            101
          </td>
        </tr>
        <tr>
          <td>
            url
          </td>
          <td>
            http://initech.zendesk.com/api/v2/organizations/101.json
          </td>
        </tr>
        <tr>
          <td>
            external_id
          </td>
          <td>
            9270ed79-35eb-4a38-a46f-35725197ea8d
          </td>
        </tr>
        <tr>
          <td>
            name
          </td>
          <td>
            Enthaze
          </td>
        </tr>
        <tr>
          <td>
            domain_names
          </td>
          <td>
            kage.com,ecratic.com,endipin.com,zentix.com
          </td>
        </tr>
        <tr>
          <td>
            created_at
          </td>
          <td>
            2016-05-21T11:10:28 -10:00
          </td>
        </tr>
        <tr>
          <td>
            details
          </td>
          <td>
            MegaCorp
          </td>
        </tr>
        <tr>
          <td>
            shared_tickets
          </td>
          <td>
            false
          </td>
        </tr>
        <tr>
          <td>
            tags
          </td>
          <td>
            Fulton,West,Rodriguez,Farley
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h2>
    SubmitTickets:
  </h2>
  <h2>
    AssignedTickets:
  </h2>
</div>
`;
